### PR TITLE
Fix: handle errors on event handlers (refs: #7629)

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Bugfix

### Purpose
- refs: https://github.com/sphinx-doc/sphinx/pull/7646#discussion_r424512792

### CODE REVIEW

File: sphinx/events.py [hunk 1]
```diff
@@ -100,11 +100,16 @@ def emit(self, name: str, *args: Any) -> List:
...
+            try:
+                if self.app is None:
+                    # for compatibility; RemovedInSphinx40Warning
+                    results.append(listener.handler(*args))
+                else:
+                    results.append(listener.handler(self.app, *args))
+            except SphinxError:
+                raise
+            except Exception as exc:
+                raise ExtensionError(exc)
```
Comment (eric-wieser) [inline sphinx/events.py]:
```suggestion
                raise ExtensionError('Handler {!r} for event {!r} threw an exception'.format(listener, event)) from exc
```
That way the chance is slightly higher that the user can recover the original stack trace.

Comment (tk0miya) [reply sphinx/events.py]: Thanks!

**Review by eric-wieser — APPROVED** [review]
Good idea

**Comment by tk0miya** [comment]
Merged. Thank you for reviewing!

File: sphinx/events.py (hunk 1 above)
Comment (eric-wieser) [reply sphinx/events.py]: Any reason for using `%` formatting instead of `.format`? Just so I can pick the right one for future contributions.

Comment (tk0miya) [reply sphinx/events.py]: Personally, I prefer it to `.format`. I think both are right.

**Pull request merged**